bbconf/modules/bedfiles.py

```diff
@@ -662,56 +723,259 @@ def update(
+                    sa_session.commit()
+                except IntegrityError as _:
+                    sa_session.rollback()
+                    _LOGGER.debug(
+                        f"Plot with name: {v.name} already exists. Updating.."
+                    )
+
+        return None
+
+    def _update_files(
+        self,
+        sa_session: Session,
+        bed_object: Bed,
+        files: BedFiles,
+        local_path: str = None,
+    ) -> None:
+        """
+        Update bed files
+
+        :param sa_session: sqlalchemy session
+        :param bed_object: bed sqlalchemy object
+        :param files: bed file files
+        """
+
+        _LOGGER.info("Updating bed files..")
+        if files:
+            files = self._config.upload_files_s3(
+                bed_object.id, files=files, base_path=local_path, type="files"
+            )
+
+        files_dict = files.model_dump(
+            exclude_defaults=True, exclude_none=True, exclude_unset=True
+        )
+        if not files_dict:
+            return None
+
+        for k, v in files:
+            if v:
+                new_file = Files(
+                    **v.model_dump(
+                        exclude_none=True,
+                        exclude_unset=True,
+                        exclude={"object_id", "access_methods"},
+                    ),
+                    bedfile_id=bed_object.id,
+                    type="file",
+                )
+
+                try:
+                    sa_session.add(new_file)
+                    sa_session.commit()
+                except IntegrityError as _:
+                    sa_session.rollback()
+                    _LOGGER.debug(
+                        f"File with name: {v.name} already exists. Updating.."
+                    )
+
+    @staticmethod
+    def _update_ref_validation(
+        sa_session: Session, bed_object: Bed, ref_validation: Dict[str, BaseModel]
+    ) -> None:
+        """
+        Update reference validation data
+
+        :param sa_session: sqlalchemy session
+        :param bed_object: bed sqlalchemy object
+        :param ref_validation: bed file metadata
+        """
+
+        if not ref_validation:
+            return None
+
+        _LOGGER.info("Updating reference validation data..")
+
+        for ref_gen_check, data in ref_validation.items():
+            new_gen_ref = GenomeRefStats(
+                **RefGenValidModel(
+                    **data.model_dump(),
+                    provided_genome=bed_object.genome_alias,
+                    compared_genome=ref_gen_check,
+                ).model_dump(),
+                bed_id=bed_object.id,
+            )
+            try:
+                sa_session.add(new_gen_ref)
+                sa_session.commit()
+            except IntegrityError as _:
+                sa_session.rollback()
+                _LOGGER.info(
+                    f"Reference validation exists for BED id: {bed_object.id} and ref_gen_check."
```

Comment:
So, this method name and doc string says that this will update reference validation data but the try/except clause shows that we can get an integrity error with error message  "Reference validation exists for BED id: ". Shouldn't we be able to overwrite it? Isn't that the point of updating it?

Reply:
Good question! For now I would keep it as it is, because I am updating bed file, and if something validation exists, I ignore it for now.
I will leave it as it is, but make some comments for it. Thanks!